packages/esm-service-queues-app/src/queue-rooms/queue-room-form.workspace.tsx

```diff
@@ -1,164 +1,169 @@
-      }
+  const onSubmit = async (data: QueueRoomFormData) => {
+    try {
+      // here doubt why queueName twice // name: string, description: string, queueUuid: string
+      await saveQueueRoom(data.queueRoomName, data.queueRoomName, data.queueRoomService);
 
-      setMissingRoomName(false);
-      setMissingQueueRoomService(false);
-
-      saveQueueRoom(queueRoomName, queueRoomName, queueRoomService).then(
-        ({ status }) => {
-          if (status === 201) {
-            showSnackbar({
-              title: t('addQueueRoom', 'Add queue room'),
-              kind: 'success',
-              subtitle: t('queueRoomAddedSuccessfully', 'Queue room added successfully'),
-            });
-            closeWorkspace();
-            mutate(`${restBaseUrl}/queueroom`);
-          }
-        },
-        (error) => {
-          showSnackbar({
-            title: t('errorAddingQueueRoom', 'Error adding queue room'),
-            kind: 'error',
-            isLowContrast: false,
-            subtitle: error?.message,
-          });
-        },
-      );
-    },
-    [queueRoomName, queueRoomService, t, closeWorkspace],
-  );
+      showSnackbar({
+        title: t('addQueueRoom', 'Add queue room'),
+        kind: 'success',
+        subtitle: t('queueRoomAddedSuccessfully', 'Queue room added successfully'),
+      });
+      closeWorkspace();
+      await mutate(`${restBaseUrl}/queueroom`);
+    } catch (error) {
+      showSnackbar({
+        title: t('errorAddingQueueRoom', 'Error adding queue room'),
+        kind: 'error',
+        isLowContrast: false,
+        subtitle: error?.message,
+      });
+    }
+  };
 
   return (
-    <Form onSubmit={createQueueRoom} className={styles.form}>
+    <Form onSubmit={handleSubmit(onSubmit)} className={styles.form}>
       <Stack gap={4} className={styles.grid}>
         <Column>
           <Layer className={styles.input}>
-            <TextInput
-              id="queueRoomName"
-              invalidText="Required"
-              labelText={t('queueRoomName', 'Queue room name')}
-              onChange={(event) => setQueueRoomName(event.target.value)}
-              value={queueRoomName}
-            />
-            {isMissingRoomName && (
-              <section>
-                <InlineNotification
-                  style={{ margin: '0', minWidth: '100%' }}
-                  kind="error"
-                  lowContrast={true}
-                  title={t('missingQueueRoomName', 'Missing queue room name')}
-                  subtitle={t('addQueueRoomName', 'Please add a queue room name')}
+            <Controller
+              control={control}
+              name="queueRoomName"
+              render={({ field }) => (
+                <TextInput
+                  {...field}
+                  id="queueRoomName"
+                  invalidText={errors.queueRoomName?.message}
+                  invalid={!!errors.queueRoomName}
+                  labelText={t('queueRoomName', 'Queue room name')}
```

Comment:
These field names should be standardized. We should either use:

- Queue service room name
- Queue service room location
- Queue Service

Or:

- Select a queue service room name
- Select a queue service location name
- Select a queue service

And not a mix of both.